### PR TITLE
enh(Listing): Add prop condition about disabling checkbox following row data

### PR DESCRIPTION
**Pull request by Thebarda**

### CODE REVIEW

File: src/Listing/index.tsx [hunk 1]
```diff
@@ -77,6 +77,7 @@ const cumulativeOffset = (element): number => {
 
 interface Props {
   checkable?: boolean;
+  checkCondition?;
```
Comment (bdauria) [inline src/Listing/index.tsx:80]:
```suggestion
  disableRowCheckCondition?;
```

File: src/Listing/index.tsx [hunk 2]
```diff
@@ -115,6 +116,7 @@ const Listing = ({
   currentPage = 0,
   totalRows = 0,
   checkable = false,
+  checkCondition,
```
Comment (bdauria) [inline src/Listing/index.tsx:119]:
```suggestion
  disableRowCheckCondition,
```

File: src/Listing/index.tsx [hunk 3]
```diff
@@ -429,6 +431,9 @@ const Listing = ({
                           inputProps={{
                             'aria-label': `Select row ${row.id}`,
                           }}
+                          disabled={
+                            checkCondition ? !!checkCondition(row) : false
```
Comment (bdauria) [inline src/Listing/index.tsx:435]:
```suggestion
                            disableRowCheckCondition.?(row) || false
```

**Comment by bdauria** [comment]
@Thebarda We might want to update the stories to add this case too

File: src/Listing/index.tsx [hunk 4]
```diff
@@ -429,6 +431,7 @@ const Listing = ({
                           inputProps={{
                             'aria-label': `Select row ${row.id}`,
                           }}
+                          disabled={!!disableRowCheckCondition(row)}
```
Comment (kduret) [inline src/Listing/index.tsx:434]: Is it needed to cast it ?

**Comment by bdauria** [comment]
@Thebarda it seems that storyshots are not updated

**Comment by Thebarda** [comment]
> @Thebarda it seems that storyshots are not updated

Updating snapshots does not change anything strangely

File: src/Listing/index.stories.tsx [hunk 5]
```diff
@@ -91,6 +91,7 @@ const Story = (props): JSX.Element => {
         rowColorConditions={rowColorConditions}
         selectedRows={listing.filter((row) => row.selected)}
         checkable
+        disableRowCheckCondition={(row): boolean => row.selected}
```
Comment (bdauria) [inline src/Listing/index.stories.tsx:94]: I think we should set the condition to something else than the row is selected

Comment (Thebarda) [reply src/Listing/index.stories.tsx:94]: Replace by what ?

**Comment by bdauria** [comment]
> > @Thebarda it seems that storyshots are not updated
> 
> Updating snapshots does not change anything strangely

Probably because of the threshold.

We can merge it like this for now, and enhence the structure of the stories later on

**Pull request merged**